config/.env.sample

```diff
@@ -2,8 +2,12 @@
 
 # Port that the server should listen on within the container (Optional)
 # PORT=4000
+# The server's domain, used for OAuth redirects (so far)
+DOMAIN=
```

Comment:
Here you have domain that CCM is running on, For local development it keep on changing. is there a way you can get this dynamically instead of hardcoding it? 
Random thought Like in the API key you have to fill where to redirect URL can we pull the domain from that?

Reply:
> Here you have domain that CCM is running on, For local development it keep on changing. is there a way you can get this dynamically instead of hardcoding it?

I have to think about this some more. I don't think there's a great solution, because the application doesn't know it's being tunneled through `ngrok`.

> Random thought Like in the API key you have to fill where to redirect URL can we pull the domain from that?

How would that work? Like I would call some Canvas API to get what I put there?

Reply:
When I say API Key that is in the Canvas Developer key space.  we need to tell canvas where to redirect after Oauth flow is done. I feel some where in the req/res i saw the redirect URL (which is our ngrok URL). You could put that in the session and use it. I have to check again but I felt that is the only way to eliminate this property in `.env`

Reply:
I will check to see what I can do. The `redirect_uri` parameter used in the OAuth authorization call has to match the redirect from the API Developer Key configuration. So the application would need to know the ngrok domain prior to making that call. Maybe it is also available via the LTI launch information.

Reply:
Okay, so the ngrok domain is tangentially include in the launch at `platformContext.targetLinkUri` (includes protocol and the ending `/lti`. However, I determined that not including the domain in the session cookie configuration causes the session integration to fail (I guess that's one of the important pieces). Because that happens on startup, I don't think there's a way to make use of launch information in this case. I'm going to consider this not possible -- unless you have other ideas.

I realize this is one more place you have to remember to update the domain (LTI Key, Developer Key, now `.env`) but I don't know what the alternative is. I guess this is some additional incentive to consider a paid `ngrok` account with persistent domains.